swig/python/CMakeLists.txt

```diff
@@ -0,0 +1,68 @@
+set(PYTHON_SWIG_BINDING sysrepo)
+include_directories(${PYTHON_INCLUDE_PATH})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(../../src/)
+set(CMAKE_SWIG_FLAGS "-c++")
+set(CMAKE_SWIG_FLAGS "-I${PROJECT_SOURCE_DIR}")
+set(CMAKE_SWIG_FLAGS "-threads")
+set(CMAKE_SWIG_OUTDIR ${CMAKE_CURRENT_BINARY_DIR})
+set(PYTHON_SWIG_MIN_VERSION "3.0.5")
+
+find_package(SWIG ${PYTHON_SWIG_MIN_VERSION})
+if (NOT SWIG_FOUND)
+  message(WARNING "Minimum SWIG version for Python is ${PYTHON_SWIG_MIN_VERSION}. Tests won't be generated.")
+endif()
+
+set_source_files_properties(${PYTHON_SWIG_BINDING}.i PROPERTIES CPLUSPLUS ON)
+
+if(${CMAKE_VERSION} VERSION_LESS "3.8.0")
+    swig_add_module(${PYTHON_SWIG_BINDING} python ${PYTHON_SWIG_BINDING}.i)
+else()
+    swig_add_library(${PYTHON_SWIG_BINDING} LANGUAGE python SOURCES ${PYTHON_SWIG_BINDING}.i)
+endif()
+swig_link_libraries(${PYTHON_SWIG_BINDING} ${PYTHON_LIBRARIES} Sysrepo-cpp)
+
+file(COPY "examples" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
+file(COPY "tests" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
```

Comment:
Is this necessary, can't examples and tests be directly compiled at the right place?

Reply:
Yes, sysrepo.py and _sysrepo.so files generated by swig are installed globally so they can be run from anywhere.